client/src/components/BlockLinkField/tests/BlockLinkField-test.js

```diff
@@ -1,305 +1,111 @@
-      const wrapper = shallow(
-        <BlockLinkField
-          {...props}
-        />
-      );
-      wrapper.setProps({
-          linkedPage: {},
-          data: { linkedPage: '/react' }
-        });
-
-      expect(wrapper.instance().getLinkedPage()).toEqual('/react');
-    });
-
-
-    it('should return the linked page within the React context', () => {
-      const wrapper = shallow(
-        <BlockLinkField
-          {...props}
-        />
-      );
-      wrapper.setProps({ linkedPage: { URLSegment: '/entwine' } });
-
-      expect(wrapper.instance().getLinkedPage()).toEqual({ URLSegment: '/entwine' });
-    });
-  });
-
-  describe('getLinkRelativeUrl()', () => {
-    it('returns relative URL for the linked page with a leading slash', () => {
-      const wrapper = shallow(
-        <BlockLinkField
-          {...props}
-        />
-      );
-
-      wrapper.instance().getLinkedPage = mockGetLinkedPage.mockReturnValue('/mockUrlWithSlash');
-      expect(wrapper.instance().getLinkRelativeUrl()).toEqual('/mockUrlWithSlash');
-
-      wrapper.instance().getLinkedPage = mockGetLinkedPage.mockReturnValue('/mockUrlWithOutSlash');
-      expect(wrapper.instance().getLinkRelativeUrl()).toEqual('/mockUrlWithOutSlash');
-    });
-
-    it('returns empty string if no relative URL provided', () => {
-      const wrapper = shallow(
-        <BlockLinkField
-          {...props}
-        />
-      );
+      },
+    })}
+    />
+  );
+  expect(container.querySelector('.block-link-field__link').textContent).toBe('/react')
+});
 
-      wrapper.instance().getLinkedPage = mockGetLinkedPage.mockReturnValue(null);
-      expect(wrapper.instance().getLinkRelativeUrl()).toEqual('');
-    });
-  });
+test('BlockLinkField should return the linked page within the React context', () => {
+  const { container } = render(
+    <BlockLinkField {...makeProps({
+      linkedPage: {
+        URLSegment: '/entwine',
+      },
+      value: '{"PageID":1}'
+    })}
+    />
+  );
+  expect(container.querySelector('.block-link-field__link').textContent).toBe('/entwine')
+});
 
-  describe('getClassNames()', () => {
-    it('fetches extra class names if provided', () => {
-      const wrapper = shallow(
-        <BlockLinkField
-          {...props}
-        />
-      );
+test('BlockLinkField returns relative URL for the linked page with a leading slash', () => {
+  const { container } = render(
+    <BlockLinkField {...makeProps({
+      linkedPage: {
+        URLSegment: 'foo',
+      },
+    })}
+    />
+  );
+  expect(container.querySelector('.block-link-field__link').textContent).toBe('/foo')
+});
 
-      wrapper.setProps({ extraClass: 'test-class' });
+test('BlockLinkField returns empty string if no relative URL provided', () => {
+  const { container } = render(
+    <BlockLinkField {...makeProps({
+      linkedPage: {},
+    })}
+    />
+  );
+  expect(container.querySelector('.block-link-field__link')).toBeNull();
+});
```

Comment:
An element with no text inside (empty string) and simply not having an element are _not_ the same thing. I accept that you can't test the internal function from the original test anymore, but we need to update the title to describe what we're actually testing now.

Reply:
Not doing